scripts/index.js

```diff
@@ -1,132 +1,132 @@
+    const objectId = event.target.id;
 
-  const h4 = document.createElement('h4');
-  h4.innerText = bookList[lastBookIndex].Title;
+    // eslint-disable-next-line max-len
+    const newBookList = this.bookList.filter((book) => book.Title !== this.bookList[objectId].Title);
 
-  const p = document.createElement('p');
-  p.innerText = bookList[lastBookIndex].Author;
+    this.bookList = newBookList;
 
-  const hr = document.createElement('hr');
+    bookDiv.innerText = '';
+    this.updateStorage();
+    this.populateBook();
+  }
 
-  const removeButton = document.createElement('button');
-  removeButton.className = 'remove-button';
-  removeButton.id = lastBookIndex;
-  removeButton.innerText = 'Remove';
+  updateStorage() {
+    if (this.storageAvailable('localStorage')) {
+      const { localStorage } = window;
+      localStorage.setItem('bookList', JSON.stringify(this.bookList));
+    }
+  }
 
-  bookDiv.appendChild(h4);
-  bookDiv.appendChild(p);
-  bookDiv.appendChild(removeButton);
-  bookDiv.appendChild(hr);
-}
+  loadFromStorage() {
+    if (this.storageAvailable('localStorage')) {
+      const { localStorage } = window;
+      const bookData = JSON.parse(localStorage.getItem('bookList'));
+      if (this.bookData !== null) {
+        this.bookList = bookData;
+        this.populateBook();
+      }
+    }
+  }
 
-function loadFromStorage() {
-  if (storageAvailable('localStorage')) {
-    const { localStorage } = window;
-    const bookData = JSON.parse(localStorage.getItem('bookList'));
-    if (bookData !== null) {
-      bookList = bookData;
-      populateBook();
+  storageAvailable(type) {
+    let storage;
+    try {
+      this.storage = window[type];
+      const x = '__storage_test__';
+      this.storage.setItem(x, x);
+      this.storage.removeItem(x);
+      return true;
+    } catch (event) {
+      return event instanceof DOMException && (event.code === 22 || event.code === 1014 || event.name === 'QuotaExceededError' || event.name === 'NS_ERROR_DOM_QUOTA_REACHED') && (storage && storage.length !== 0);
     }
   }
-}
 
-function addBook() {
-  const bookTitle = document.getElementsByClassName('book-title')[0];
-  const author = document.getElementsByClassName('author')[0];
+  addBook() {
+    const bookTitle = document.getElementsByClassName('book-title')[0];
+    const author = document.getElementsByClassName('author')[0];
 
-  if (bookTitle.value === '' || author.value === '') {
-    // eslint-disable-next-line no-alert
-    alert('Please make sure to fill both Title & Author fields!');
-  } else {
-    bookList.push({
-      Title: bookTitle.value,
-      Author: author.value,
-    });
+    if (bookTitle.value === '' || author.value === '') {
+      // eslint-disable-next-line no-alert
+      alert('Please make sure to fill both Title & Author fields!');
+    } else {
+      this.bookList.push({
+        Title: bookTitle.value,
+        Author: author.value,
+      });
```

Comment:
- [ ] Adding a book to the collection is not working, When I'm trying to add a book it shows this error on the console. Please fix this issue and make sure to maintain all of the original functionality of the application.

![image](https://user-images.githubusercontent.com/83803414/183675623-892b7dc2-0c47-402b-b06c-eab99484a254.png)

Reply:
Hi Ranjeet, I'm unable to replicate your issue. Could you walk me through what exactly has caused the error? On my end I'm able to add and remove with no console errors

Reply:
@WalterOkumu I think It's because of the localstorage. The object wasn't available on my browser.

Reply:
Found it, it was the constructor, bookList wasn't initialised to be an array

Reply:
@WalterOkumu good job 👍